### DISCUSSION TITLE
openthread-br on openwrt 23.05.03 thread device no fd11 address

### DISCUSSION THREAD

**Discussion by kaizirlewagen** [openthread/openthread#10002] [Q&A]
I use openthread-br with OpenWrt 23.05.03. I can compile the firmware with openthread-br. I can create a thread network and i can join a device (nordic devices).

```
ot-ctl dataset init new
ot-ctl dataset commit active
ot-ctl prefix add fd11:22::/64 pasor
ot-ctl ifconfig up
ot-ctl thread start
ot-ctl netdata register
ot-ctl dataset active -x
```
```
ot-ctl ipaddr
2001:db8:85a3:0:3ab8:170b:588e:6344
fdec:63c6:ed46:cb31:0:ff:fe00:fc31
fdec:63c6:ed46:cb31:0:ff:fe00:fc10
fd11:22:0:0:5994:6da0:314f:3073
fdec:63c6:ed46:cb31:0:ff:fe00:fc00
fdec:63c6:ed46:cb31:0:ff:fe00:6400
fdec:63c6:ed46:cb31:7dcc:f440:a2e:c5b8
fe80:0:0:0:18e1:5964:516c:ebdd
Done
```

But the cli device does not get an fd11 address. The logs does not raise an error.

```
> ipaddr
fdde:ad00:beef:0:0:ff:fe00:fc00
fdde:ad00:beef:0:0:ff:fe00:2c00
fdde:ad00:beef:0:1c54:b2c5:20d8:4d93
fe80:0:0:0:6468:e0be:d327:1058
Done
```

Someone any idea where the error could be? I try to check all installation tips, but i don't find my error.

Thx, Tom

**Comment by jwhui** [comment]
Does your CLI device have SLAAC enabled?

https://github.com/openthread/openthread/blob/223935bf0c6bd02675efcb56254d21abaa1fb530/src/core/config/ip6.h#L130

**Comment by kaizirlewagen** [comment]
I don't think so. I use the firmware for the nordic device from https://github.com/openthread/ot-nrf528xx.
I am new to the openthread project, so must SLAAC enabled through the openthread Makefile? Or while building the firmware for the nordic device?

I test the openthread on raspbian and all functions like routing, join, nat64 are working fine... the problems are only with openwrt. So, i don't find the error on openwrt. For example, missing dependencies while building the openwrt image?

**Reply by wgtdkp** [reply]
```
> ipaddr
fdde:ad00:beef:0:0:ff:fe00:fc00
fdde:ad00:beef:0:0:ff:fe00:2c00
fdde:ad00:beef:0:1c54:b2c5:20d8:4d93
fe80:0:0:0:6468:e0be:d327:1058
Done
```

It looks like your CLI device has a different mesh-local prefix than the BR (fdde:ad00:beef:0::/64 vs fdec:63c6:ed46:cb31::/64). I think your CLI device is not in the same Thread network of the BR. You cna verify this by pinging the BR address from the CLI device:

```
> ping fdec:63c6:ed46:cb31:7dcc:f440:a2e:c5b8
```

You should get no reply.

To get your CLI device into the BR network, follow steps in https://openthread.io/codelabs/openthread-border-router#join-the-otbr-network

**Comment by kaizirlewagen** [comment]
I do the joining with the following commands at border router:

```
$ ot-ctl commissioner start
Done

$ ot-ctl commissioner state
Active
Done

$ ot-ctl commissioner joiner add "*" J01NME
```

After that i switch to the cli device and do:

```
> factoryreset
> ifconfig up
> joiner start J01NME
> thread start
```

This works on border router with raspbian but not with openthread on openwrt 23.05.03 system. So, are the commands wrong?

@wgtdkp Your right the ping does not work. I have to check the params i use to build the firmware for the nordic devices. 

https://openthread.io/codelabs/openthread-hardware?hl=de#4

Maybe here is something wrong. But i don't find the error why the devices works with raspbian but not with openwrt.

**Comment by kaizirlewagen** [comment]
The problem with the missing fd11:22::/64 address has the origin how the thread network is created.

not working:

```
ot-ctl dataset init new
ot-ctl dataset commit active
ot-ctl prefix add fd11:22::/64 pasor
ot-ctl ifconfig up
ot-ctl thread start
ot-ctl netdata register
```

working:
```
ot-ctl dataset init new
ot-ctl dataset channel 11
ot-ctl dataset channelmask 0x7fff800
ot-ctl dataset panid 0xfb00
ot-ctl dataset extpanid fb02fb02fb020000
ot-ctl dataset networkkey 00112233445566778899aabbccddeeff
ot-ctl dataset meshlocalprefix fd00:0000:fb01:0001::
ot-ctl dataset networkname WSN
ot-ctl dataset commit active
ot-ctl ifconfig up
ot-ctl thread start
ot-ctl prefix add fd11:22::/64 pasor
ot-ctl netdata register
```

```
> ipaddr
fd00:0:fb01:1:0:ff:fe00:3000
fd11:22:0:0:23fe:e5f:c4ee:cd7a
fd00:0:fb01:1:6c5f:8d8f:da20:a8af
fe80:0:0:0:6468:e0be:d327:1058
Done
```

So, i have to check the single params which one fixes the missing address.

**Reply by jwhui** [reply]
I suspect that the Active Operational Datasets are not consistent across your devices.

Try running `dataset active` and comparing the result across your devices.
